### PR TITLE
Add support for ajv-keywords

### PR DESCRIPTION
**Pull request by techntools**
Useful to add custom validations.

I am using it in [sequelize-to-openapi](https://github.com/techntools/sequelize-to-openapi)

### CODE REVIEW

**Comment by techntools** [comment]
@wesleytodd PTAL

**Comment by wesleytodd** [comment]
Hey! Can you describe a bit what this does? I am not familiar with the lib so I would like to understand a bit more what this helps folks do before taking on the additional dep.

File: index.js [hunk 1]
```diff
@@ -63,11 +63,11 @@ module.exports = function ExpressOpenApi (_routePrefix, _doc, _opts) {
   }
 
   // Validate path middleware
-  middleware.validPath = function (schema = {}) {
+  middleware.validPath = function (schema = {}, pathOpts = {}) {
     let validate
     function validSchemaMiddleware (req, res, next) {
       if (!validate) {
-        validate = makeValidator(middleware, getSchema(validSchemaMiddleware), opts)
+        validate = makeValidator(middleware, getSchema(validSchemaMiddleware), { ...pathOpts, ...opts })
```
Comment (wesleytodd) [inline index.js:70]: If this is just merging them back together could we just drop the new additional argument?

File: lib/validate.js [hunk 2]
```diff
@@ -93,6 +94,8 @@ module.exports = function makeValidatorMiddleware (middleware, schema, opts) {
         strict: opts.strict === true ? opts.strict : false
       })
       addFormats(ajv)
+
+      if (opts.keywords) { addKeywords(ajv, opts.keywords) }
```
Comment (wesleytodd) [inline lib/validate.js:98]: Is there a reason not to always load this? Is it a perf hit?

File: index.js (hunk 1 above)
Comment (wesleytodd) [reply index.js:70]: Oh wait, `opts` here is from somewhere else. Ah, I see the reason now. Sorry, first week back from a month of vacation so my brain is not fully back in a detail oriented mode.

**Comment by wesleytodd** [comment]
I am going to mark this as approved since it requires it to run CI for first time contributors, but I still would like to hear more about what this provides functionality wise, and maybe even ask that we write slightly more docs on it since that package doesn't even have a good description in it's readme.

**Review by wesleytodd — DISMISSED** [review]

File: lib/validate.js (hunk 2 above)
Comment (techntools) [reply lib/validate.js:98]: Needed only when custom keywords are used in api definition to validate request.

Its not a perf hit.

**Comment by techntools** [comment]
Lets say you have a following json schema definition:

```js
{
    type: 'object',
    properties: {
        name: {
            type: 'string',
            pattern: '^[a-zA-Z]$'
        }
    }
}
```

But json schema does not support flags such as i, g, etc.

With ajv-keywords, you can use custom keyword like ```regexp``` to get over that limitation:

```js
{
    type: 'object',
    properties: {
        name: {
            type: 'string',
            regexp: {
                pattern: '^[a-z]$',
                flags: 'i'
            }
        }
    }
}
```

**Comment by wesleytodd** [comment]
Ah, interesting and for sure useful. Does this then also hook into the validation so that works correctly? Sorry for the niave questions here, but I think that this would be great to add to your docs if that is something you could do. Also, I think now that I have approved, if you push again it should re-triger the CI.

**Comment by techntools** [comment]
@wesleytodd 

It does participate in the validation

I have added an example as well with ```validPath()```

**Comment by techntools** [comment]
@wesleytodd 

I apologise that your approval got dismissed as I forced pushed to my branch.

Added more test to show that query param can be validated with ajv-keywords

**Review by wesleytodd — APPROVED** [review]
New approval, but I think maybe the GH feature here is broken because what use is it if a new push doesn't start triggering the ci? So if this doesn't work I might go ahead and change the setting on the repo.

**Comment by techntools** [comment]
@wesleytodd 

All checks are passed

Thanks for updating the CI.

Please approve once again.

**Review by wesleytodd — APPROVED** [review]

**Pull request merged**